runtime/src/qio/qio.c

```diff
@@ -4388,6 +4388,16 @@ qioerr qio_get_fs_type(qio_file_t* fl, int* out)
   return 0;
 }
 
+qioerr qio_get_fd(qio_file_t* fl, int* out)
+{
+  if(fl->fd != -1)
```

Comment:
let's make this condition also check fl != NULL

Reply:
made the changes